src/lib/hash.c

```diff
@@ -79,18 +79,19 @@
 #include <rnp/rnp_sdk.h>
 #include <stdio.h>
 
-static pgp_map_t hash_alg_map[] = {
-  {PGP_HASH_MD5, "MD5"},
-  {PGP_HASH_SHA1, "SHA1"},
-  {PGP_HASH_RIPEMD, "RIPEMD160"},
-  {PGP_HASH_SHA256, "SHA256"},
-  {PGP_HASH_SHA384, "SHA384"},
-  {PGP_HASH_SHA512, "SHA512"},
-  {PGP_HASH_SHA224, "SHA224"},
-  {PGP_HASH_SM3, "SM3"},
-  {PGP_HASH_CRC24, "CRC24"},
-  {0x00, NULL}, /* this is the end-of-array marker */
-};
+static const struct hash_alg_map_t {
+    uint8_t     type;
```

Comment:
This should be `pgp_hash_alg_t` so that the compiler has a chance to catch issues (at least clang seems to be smart enough).